hq/app/schedule/IamAudit.scala

```diff
@@ -1,30 +1,56 @@
 package schedule
 
-import com.gu.anghammarad.models.{Target, AwsAccount => Account}
+import com.gu.anghammarad.models.{Notification, Target, AwsAccount => Account}
 import config.Config.{iamAlertCadence, iamHumanUserRotationCadence, iamMachineUserRotationCadence}
 import logic.DateUtils
 import model._
 import org.joda.time.{DateTime, Days}
 import play.api.Logging
 import schedule.IamMessages._
-import schedule.IamNotifier.createNotification
+import schedule.IamNotifier.notification
 import utils.attempt.FailedAttempt
 
 object IamAudit extends Logging {
 
-  def makeIamNotification(flaggedCredsToNotify: Map[AwsAccount, Seq[IAMAlertTargetGroup]]): List[IamNotification] = {
+  def makeNotification(flaggedCredsToNotify: Map[AwsAccount, Seq[IAMAlertTargetGroup]]): List[IamNotification] = {
     flaggedCredsToNotify.toList.flatMap { case (awsAccount, targetGroups) =>
       if (targetGroups.isEmpty) {
         logger.info(s"found no IAM user issues for ${awsAccount.name}. No notification required.")
         None
       } else {
-        targetGroups.map { tg =>
-          logger.info(s"for ${awsAccount.name}, generating iam notification message for ${tg.users.length} user(s) with outdated keys and and/or missing mfa")
-          val (usersToReceiveWarningAlerts, usersToReceiveFinalAlerts) = sortUsersIntoWarningOrFinalAlerts(tg.users)
-          createNotifications(warning = true, usersToReceiveWarningAlerts, awsAccount, tg.targets) ++ createNotifications(warning = false, usersToReceiveFinalAlerts, awsAccount, tg.targets)
+        targetGroups.map{ tg =>
+          val (warningNotification, finalNotification) = createWarningAndFinalNotification(tg, awsAccount)
```

Comment:
I think these 3 lines could be neater  if you made the return type of `createWarningAndFinalNotification` an `IamNotification` and make it take `users` as a parameter - this avoids the awkward (Option, Option) tuple return type

Reply:
This is a great idea, thanks!